### ISSUE TITLE
Error when http response already returned

### ISSUE THREAD

**Issue by karlatkinson** [ekmsystems/serilog-enrichers-correlation-id#3]
Hi, we started using this in a web api solution.
There is an issue where by if the response has already returned by the time serilog comes to enrich the log message, it can error on adding the correlation id to the response on this line:

https://github.com/ekmsystems/serilog-enrichers-correlation-id/blob/master/src/Serilog.Enrichers.CorrelationId/Enrichers/CorrelationIdHeaderEnricher.cs#L36

This is possibly due to the way we've configured our pipeline or something but seems like it should handle it rather than throwing an exception.
We've fixed this with the following (can raise a PR if you want).

```
if (!HttpContext.Current.Response.HeadersWritten)
{
   HttpContext.Current.Response.AddHeader(_headerKey, correlationId);
}
```

**Comment by mrstebo** [comment]
That is true. Yeah, would be preferable if it didn't just fall over. 
I'd submit a PR and see what the guys think. Maybe it could be an issue with where this enricher is in your pipeline? You'd supposedly want the correlation ID in all your responses, right?

**Comment by karlatkinson** [comment]
@mrstebo cool, raised a PR
I think we're seeing this due to our combination of using a ExceptionFilterAttribute to return a HttpResponseException on an exception and https://github.com/serilog-web/classic which logs requests using a HttpModule

**Comment by ejcoyle88** [comment]
Hi, thanks for reporting this and sorry it took forever to get around to it.

I've now replicated the change you made but in the new codebase, and released it as [v3.0.0](https://github.com/ekmsystems/serilog-enrichers-correlation-id/releases/tag/v3.0.0). I decided to do a major version bump because of the framework version bump.

Any other issues, please let us know.

**Comment by CreAim-EvL** [comment]
Hi, I noticed this is also an issue for .net core.
and  in order to make the similar change in .NET Core the property to check (on the Response, prior to adding the header) is called  "HasStarted"  instead of "HeadersWritten"
